helios-services/src/main/java/com/spotify/helios/agent/Supervisor.java

```diff
@@ -375,7 +385,13 @@ public void perform(final boolean done) throws InterruptedException {
           .setMaxIntervalMillis(SECONDS.toMillis(30))
           .build().newScheduler();
 
-      // Kill the container after stopping the runner
+
+      // TODO(negz): Use Docker stop instead of kill?
+      // I assume this loop is intended to handle the case where the above runner.stop() fails to
+      // communicate with the Docker API, because runner.stop() already sends a SIGKILL if
+      // necessary. What good could sending more SIGKILLs do if the first signal was sent
+      // successfully? If we *are* working around Docker API failures why not send more SIGTERMs
+      // to give the container the chance to shutdown gracefully?
```

Comment:
I'm tempted to address this TODO as a separate commit in this PR, but I'm not quite sure about the current implementation.

Do we ask the runner to stop the container, but then fall back to killing it in the supervisor because we want to handle the case where the container is somehow running but its runner is not? If that's not a concern it might be nicer to move the retry logic into the runner and just have it send a bunch of `docker.stopContainer()`s with the requested SIGKILL deadline.